test/vanilla/legacy/AcceptanceTests/conftest.py

```diff
@@ -85,3 +86,32 @@ def on_response(self, request, response, **kwargs):
 def cookie_policy():
     return CookiePolicy()
 
+@pytest.fixture()
+def credential():
+    """I actually don't need anything, since the authentication policy
+    will bypass it.
+    """
+    class FakeCredential:
+        pass
+    return FakeCredential()
+
+@pytest.fixture()
+def authentication_policy():
+    from azure.core.pipeline.policies import SansIOHTTPPolicy
+    return SansIOHTTPPolicy()
+
+class AuthenticationPolicyToken(BearerTokenCredentialPolicy):
```

Comment:
this isn't PR blocking, but I don't think we want to override the credential policy in the tests, that negates some of what we're trying to test. Instead, I think you can update line 94's `FakeCredential` to have a method called `get_token()` (following the protocol of a [`TokenCredential`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/credentials.py))  that returns something deterministic, i.e. `AccessToken("AAD", random expires time). That way, we're using the generated policies, which more accurately tests

Reply:
@iscai-msft, I find that `enforce_https=False` could skip `_enforce_https` in `BearerTokenCredentialPolicy`. Now I have changed to override `get_token` instead of override auth policy. For the `get_token`, if our target is to check credential scopes, I think there is no need to use jwt to encode it. Instead, we could check it directly.